packages/x-license/src/Watermark/Watermark.tsx

```diff
@@ -13,6 +13,8 @@ function getLicenseErrorMessage(licenseStatus: LicenseStatus) {
       return 'MUI X Invalid license key';
     case LICENSE_STATUS.OutOfScope:
       return 'MUI X License key plan mismatch';
+    case LICENSE_STATUS.ProductNotCovered:
+      return 'MUI X Product not covered by plan';
```

Comment:
I advocated to change `ProductNotCovered` into a more specific `NotAvailableInInitialProPlan` in #13568, so that we can display a more precise error in the doc.
We could indeed rename `OutOfScope` which is super broad right now.

Reply:
> `NotAvailableInInitialProPlan` so that we can display a more precise error in the doc.

It feels like this should be one error, with metadata. There will be more cases like this in the future of having a paid for one of the MUI X plans, and trying to use a different plan. I don't see us creating a new error type each time.

Reply:
> I don't see creating a new error type each time scaling.

The status is only used to do some sort of switch and display a warning message.
So I don't see how metadata would be more or less scalable than a different status message.
Even if we end up with 20 different statuses, the big part is the switch of conditions.